libr/socket/run.c

```diff
@@ -475,6 +475,100 @@ R_API const char *r_run_help() {
 	"# nice=5\n";
 }
 
+static int fd_forward (int in_fd, int out_fd, char **buff) {
+	int size = 0;
+
+	if (ioctl (in_fd, FIONREAD, &size) == -1) {
+		perror ("ioctl");
+		return -1;
+	}
+
+	if (size == 0) // child process exited or socket is closed
+		return -1;
+
+	*buff = realloc (*buff, size);
```

Comment:
dont realloc over the same buffer or memleaks can happen. use a temporary variable and check for null before going further